Fix hit error bar icon orientation

outfoxxed opened the pull request:
Icons now keep their original orientation when the hit error bar is flipped, instead of breaking.

I probably have every possible flip/rotation here,

Before PR:
![image](https://user-images.githubusercontent.com/83010835/196028794-c2e1e83b-bdee-490a-8f4f-8a7ef8efbcc0.png)

After PR:
![image](https://user-images.githubusercontent.com/83010835/196028841-e65040f7-b718-495d-b13e-1bf2b0f9ffd1.png)

Notes:
After running formatting and style analysis (fairly sure thats what `InspectCode.sh` is supposed to do) nearly every file was modified, so I just copied out the formatted version of my change, reverted, and re-applied the change.

Not sure if the new object allocation per `Update()` is bad, didn't check if it runs per frame. Maybe all of this should go into the `set` blocks for `Rotation` and `Scale`.

outfoxxed committed:
Fix hit error bar icon orientation

Icons now keep their original orientation when the hit error bar is
flipped

bdach commented:
Please use [`UprightAspectMaintainingContainer`](https://github.com/ppy/osu/blob/master/osu.Game/Graphics/Containers/UprightAspectMaintainingContainer.cs) instead of free-rolling another local implementation of it.

outfoxxed committed:
Reuse existing code in hit error bar fix

Use `UprightAspectMaintainingContainer` instead of a local fix for
hit error bar orientation

bdach commented on osu.Game/Screens/Play/HUD/HitErrorMeters/BarHitErrorMeter.cs:
```diff
@@ -273,45 +274,74 @@ private void recreateLabels(LabelStyles style)
 
                 case LabelStyles.Icons:
-                    labelEarly = new SpriteIcon
+                    labelEarly = new UprightAspectMaintainingContainer
```
The fact that this is created in four places within a switch statement puts me off a bit. The cleaner/better way to do this, in my opinion, would be to create the `UprightAspectMaintainingContainer`s in BDL, add them once, and only clear or replace their children in `recreateLabels()`, rather than recreate the entire container every time the setting changes.

outfoxxed committed:
Only create hit error bar icon containers once

Move icon container creation to BDL, and only recreate the contained
icon when `recreateLabels` is called

peppy committed:
Fix formatting issues
